### DISCUSSION TITLE
Accessing the best validation loss so far in validation_epoch_ends

### DISCUSSION THREAD

**Discussion by jandonov** [Lightning-AI/pytorch-lightning#6628] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
How can I access the best validation loss in `validation_epoch_end`?  I am assuming the loss is stored somewhere as it is used to save the best model based on it, so I was wondering if I can somehow directly access it instead of tracking it my self.

**Comment by kkirtac** [comment, marked as answer]
I think if you include a [ModelCheckpoint](https://pytorch-lightning.readthedocs.io/en/latest/extensions/generated/pytorch_lightning.callbacks.ModelCheckpoint.html#pytorch_lightning.callbacks.ModelCheckpoint) in your trainer, you should be able to retrieve the [best model score](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/callbacks/model_checkpoint.py#L201). But you should set your validation loss as the monitored variable in your ModelCheckpoint callback. From your `validation_epoch_end`, you should be able to retrieve your best score by reaching the callback with something similar to `self.trainer.callback.best_model_score`.

**Reply by jandonov** [reply]
Thanks this seems like a logical solution. Will test it and see how it works :)
